### PR TITLE
Tweak log levels

### PR DESCRIPTION
**Pull request by enyst**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

This PR proposes a few tweaks for log levels:
- display once at info the action and the obs (ref [slack](https://openhands-ai.slack.com/archives/C06P5NCGSFP/p1730486944028619?thread_ts=1730486459.361769&channel=C06P5NCGSFP&message_ts=1730486944.028619))
- send to debug channel most of logging in `github.py` since that is not active in most use cases and on local it sounds like something's wrong.

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:1aa983c-nikolaik   --name openhands-app-1aa983c   docker.all-hands.dev/all-hands-ai/openhands:1aa983c
```

### CODE REVIEW

**Review by tofarr — APPROVED** [review]
LGTM

**Review by rbren — CHANGES_REQUESTED** [review]
I explicitly do not want actions and observations in the info logs--will follow up

**Comment by rbren** [comment]
@enyst we need to keep actions and observations out of the default `info` logs, because they can contain (a) a LOT of information, and (b) private information

I'd support having an env var that swaps them from debug into info though, like `LOG_ALL_EVENTS`

**Comment by enyst** [comment]
> @enyst we need to keep actions and observations out of the default `info` logs, because they can contain (a) a LOT of information, and (b) private information

Okay. Oh, this is an issue on the hosted version, I suppose? On local it looks really strange to not tell the user _anything_.

Env var sounds good.

**Comment by rbren** [comment]
Yeah I'd say it's an issue in any kind of multi-tenant environment.

Also, ideally, the UI would give you all this information, and the server logs should really just be there to know "is the server functioning properly." I think right now it's kind of a hacky way to gain the level of agent observability we _want_ in the UI but don't have.

**Comment by enyst** [comment]
On further thought, we already have other env vars changing the logging behavior, like LOG_TO_FILE and SKIP_CONTAINER_LOGS somewhere, adding another is not helping predictability. FWIW maybe I didn't lose track yet, but it's going to happen. 😅

Does this work though as a temporary measure? I think we need to think this over.

File: openhands/controller/agent_controller.py [hunk 1]
```diff
@@ -164,6 +165,10 @@ def log(self, level: str, message: str, extra: dict | None = None):
         Args:
             message (str): The message to log.
         """
+        # override log level to 'info' if LOG_ALL_EVENTS is set and level is 'debug'
+        if os.getenv('LOG_ALL_EVENTS') in ('true', '1') and level.lower() == 'debug':
```
Comment (rbren) [inline openhands/controller/agent_controller.py:169]: probably this switch should go into the place where we log the event. There might be other debug logs coming through here that the user doesn't need to see

**Review by rbren — APPROVED** [review]
Unblocking but left one comment!

File: openhands/controller/agent_controller.py (hunk 1 above)
Comment (enyst) [reply openhands/controller/agent_controller.py:169]: Well, currently it's fine, but yes. We are not going to remember for the new ones that there's a trick. 😅

**Pull request merged**